numpy/lib/stride_tricks.py

```diff
@@ -111,6 +112,164 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def _sliding_window_view_dispatcher(x, shape, axis=None, *,
+                                    subok=None, writeable=None):
+    return (x,)
+
+
+@array_function_dispatch(_sliding_window_view_dispatcher, module='numpy')
+def sliding_window_view(x, shape, axis=None, *, subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given shape.
+
+    Creates a sliding window view of the N dimensional array with the given
+    window shape. Window slides across each dimension of the array and extract
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    shape : int or tuple of int
+        The shape of the window. If `axis` is not present, must have same
+        length as the number of input array dimensions.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    axis : int or tuple of int, optional
+        If present, `shape[i]` will refer to the axis `axis[i]` of `x`,
+        otherwise `shape[i]` will refer to axis `i` of `x`.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array.
```

Comment:
Two question here:

- Where do the sliding window dimensions get inserted -- at the start or at the end?
- Can you specify what the shape of the result will be, in terms of the original shape of the array?

Reply:
Good points! Will update docstring. :+1:

Reply:
I tried to clarify the docstring. As a result, I also renamed the `shape` argument to `window_shape`. I am not 100% pleased with the results of my efforts but didn't manage to come up with a better formulation.

Suggestions are very welcome!